crates/lang/ir/src/ir/attrs.rs

```diff
@@ -764,6 +788,18 @@ impl TryFrom<syn::NestedMeta> for AttributeFrag {
                             }
                             return Err(format_err!(name_value, "expecteded 4-digit hexcode for `selector` argument, e.g. #[ink(selector = 0xC0FEBABE]"))
                         }
+                        if name_value.path.is_ident("label") {
+                            if let syn::Lit::Str(lit_str) = &name_value.lit {
+                                let name = lit_str.value();
```

Comment:
This needs a check to be a valid Rust identifier.
You can check this by using
```rust
syn::parse_str::<syn::Ident>()
    .unwrap_or_else(|error| {
        panic!("encountered invalid non-identifier namespace argument: {}", error)
    })
```

Reply:
Actually instead of panicking you should return a proper `syn::Error` via `format_err!` instead.

Reply:
https://github.com/paritytech/ink/pull/931/files#diff-024136e95afaa59cacb0e87a4287e1e425de563db108032ba4b2d40df2c160eaR770

Reply:
@Robbepop  I'm waiting for your decision in the [thread](https://github.com/paritytech/ink/pull/923#discussion_r714503215).
Because it will affect this part too=)

Reply:
I cannot tell either what the better design is.
Let us summarize what you propose:

1) `#[ink(label = "foo")]` on an ink! implementation block is a prefix for `#[ink(label = "bar")]` on ink! messages in that same block. So both combined you end up with a label `foo::bar`.
2) Any `#[ink(label = "bar")]` on an ink! message or constructor in an ink! implementation block will overwrite the label of the ink! message or constructor entirely. Any `#[ink(label = "foo")]` on the ink! implementation block will be silently ignored.

Reply:
1. Yes, the label on ink! the message will overwrite only suffix(doesn't affect the prefix from the label on ink! impl block).
2. Yes, the label on ink! the message will overwrite entirely(affect the prefix from the label on ink! impl block).
3. `#[ink(label = "foo")]` on implementation block will be assumed as an error(it means remove support of the label for impl block).  `#[ink(label = "bar")]` on an ink! message in ink! implementation block will overwrite the default behavior(when the label of ink! message is a combination of trait name and message label).

I implemented the first case.
For me, the second case seems similar to the behavior of the `selector` attribute(if you defined the `selector`, it will ignore `namespace`.
The third case I created after your idea to remove the label for ink! implementation block=)

The idea of the `label` attribute is to add the ability to change the generated ABI. 
For example, in the code, I have two traits: One for internal usage(without ink! messages); another with ink! messages. Both traits have the method with the same name. It requires the developer in the code to explicitly specify the trait which he wants to use. To avoid this, I want to change the name of the method in the trait with ink! messages(I can't change the name in internal trait). But I will change the label of the method in ABI. In this case, the `label` attribute will help me to generate the right ABI.